buildkite/resource_organization_settings.go

```diff
@@ -3,114 +3,262 @@ package buildkite
 import (
 	"context"
 	"fmt"
+	"log"
 	"strings"
 
-	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	resource_schema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
-func resourceOrganizationSettings() *schema.Resource {
-	return &schema.Resource{
-		CreateContext: CreateUpdateDeleteOrganizationSettings,
-		ReadContext:   ReadOrganizationSettings,
-		UpdateContext: CreateUpdateDeleteOrganizationSettings,
-		DeleteContext: DeleteOrganizationSettings,
-		Importer: &schema.ResourceImporter{
-			State: schema.ImportStatePassthrough,
-		},
-		Schema: map[string]*schema.Schema{
-			"uuid": {
-				Type:     schema.TypeString,
+type organizationResourceModel struct {
+	AllowedApiIpAddresses types.List   `tfsdk:"allowed_api_ip_addresses"`
+	ID                    types.String `tfsdk:"id"`
+	UUID                  types.String `tfsdk:"uuid"`
+}
+
+type organizationResource struct {
+	client *Client
+}
+
+func newOrganizationResource() resource.Resource {
+	return &organizationResource{}
+}
+
+func (organizationResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_organization_settings"
```

Comment:
Given the `datasource` is just called `organization`, would we not do the same here?

Reply:
Yeah thats right. The new resource built using the framework (this one) is `buildkite_organization` and we will leave the old one around using sdkv2 named `buildkite_organization_settings`. After some deprecation period, we can remove it. Lets say by version v0.21.0 - this will need documenting as well